src/Helper/Order.php

```diff
@@ -0,0 +1,122 @@
+<?php declare(strict_types=1);
+
+namespace Satispay\Helper;
+
+use Satispay\Handler\PaymentHandler;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsAnyFilter;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\RangeFilter;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Sorting\FieldSorting;
+use Satispay\System\Config as SatispayConfig;
+
+class Order
+{
+    /**
+     * @var EntityRepositoryInterface
+     */
+    protected $orderRepository;
+    /**
+     * @var EntityRepositoryInterface
+     */
+    protected $paymentRepository;
+    /**
+     * @var SatispayConfig
+     */
+    protected $config;
+
+    /**
+     * Order constructor.
+     * @param EntityRepositoryInterface $orderRepository
+     * @param EntityRepositoryInterface $paymentRepository
+     */
+    public function __construct(
+        SatispayConfig $config,
+        EntityRepositoryInterface $orderRepository,
+        EntityRepositoryInterface $paymentRepository
+    ) {
+        $this->config = $config;
+        $this->orderRepository = $orderRepository;
+        $this->paymentRepository = $paymentRepository;
+    }
+
+    /**
+     * Return the Satispay orders following a complex criteria rule
+     *
+     * @param $context
+     * @return array|\Shopware\Core\Framework\DataAbstractionLayer\EntityCollection
+     */
+    public function getSatispayOrders($context, $salesChannelId)
+    {
+        $satispayPaymentId = $this->getSatispayPaymentId($context);
+        if ($satispayPaymentId == null) {
+            return [];
+        }
+        $rangeStart = $this->getStartDateScheduledTime($salesChannelId);
+        $rangeEnd = $this->getEndDateScheduledTime();
+        // create a complex criteria condition
+        $criteriaOrder = new Criteria();
```

Comment:
To clean a bit the code can you extract the criteria building in a method? @MichaelBattistonBitbull